src/deepmol/utils/decorators.py

```diff
@@ -0,0 +1,91 @@
+from typing import Union
+
+
+def inplace_decorator(method: callable) -> Union[callable, None]:
+    """
+    Decorator to make a method inplace.
+
+    Parameters
+    ----------
+    method: callable
+        Method to decorate.
+
+    Returns
+    -------
+    inplace_method: callable
+        Decorated method.
+    """
+    def inplace_method(self, *args, inplace=False, **kwargs):
+        """
+        Method to make inplace.
+
+        Parameters
+        ----------
+        self: object
+            Object to apply the method to.
+        args: list
+            Arguments to pass to the method.
+        inplace: bool
+            Whether to apply the method inplace.
+        kwargs: dict
+            Keyword arguments to pass to the method.
+
+        Returns
+        -------
+        result: object
+            Result of the method.
+        """
+        if inplace:
+            method(self, *args, **kwargs)
+            return None
+        else:
+            result = self.__copy__()
+            method(result, *args, **kwargs)
+            return result
+    return inplace_method
+
+
+def copy_on_write_decorator(method: callable) -> Union[callable, None]:
```

Comment:
This name sounds strange... I understood what it does but I had to read the code. Can you please describe a little more in-depth what it does to the method in the documentation and assign a more self-explanatory name? I understand that the featurizers use this decorator because they are changing another object and not the self as in **inplace_decorator**.